sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -436,4 +436,12 @@ namespace Azure { namespace Core { namespace Test {
     CheckBodyFromBuffer(*response, expectedResponseBodySize);
   }
 
+  TEST_F(TransportAdapter, requestFailedException)
```

Comment:
Add a test that you can catch the runtime_error and dynamic cast to the RequestFailedException